coderd/database/dbfake/dbfake.go

```diff
@@ -446,12 +446,37 @@ func (q *FakeQuerier) getLatestWorkspaceBuildByWorkspaceIDNoLock(_ context.Conte
 func (q *FakeQuerier) getTemplateByIDNoLock(_ context.Context, id uuid.UUID) (database.Template, error) {
 	for _, template := range q.templates {
 		if template.ID == id {
-			return template.DeepCopy(), nil
+			return q.templateWithUser(template), nil
 		}
 	}
 	return database.Template{}, sql.ErrNoRows
 }
 
+func (q *FakeQuerier) templatesWithUser(tpl []database.TemplateTable) []database.Template {
+	cpy := make([]database.Template, 0, len(tpl))
+	for _, t := range tpl {
+		cpy = append(cpy, q.templateWithUser(t))
+	}
+	return cpy
+}
+
+func (q *FakeQuerier) templateWithUser(tpl database.TemplateTable) database.Template {
+	var user database.User
+	for _, _user := range q.users {
```

Comment:
Should we hold the mutex or rename the method? IMO it should be clear that we either need to or not.

Reply:
I can make out more obvious, it should only ever be called from a locked context.